codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parsers/JsonSerializerGenerator.kt

```diff
@@ -321,7 +324,12 @@ class JsonSerializerGenerator(protocolConfig: ProtocolConfig) : StructuredDataSe
         val keyName = safeName("key")
         val valueName = safeName("value")
         rustBlock("for ($keyName, $valueName) in ${context.valueExpression.asRef()}") {
-            serializeMember(MemberContext.mapMember(context, keyName, valueName))
+            val keyTarget = model.expectShape(context.shape.key.target)
```

Comment:
not that it totally matters, but you can probably avoid this by going through the `memberSerializer` for `key` rather than hard coding it. Then you could even have a blob key or something really crazy and it would work fine.

Reply:
That sounds ideal, but things are more muddy than that since `serializeMember` generates code to exercise the JsonValueWriter, which we definitely wouldn't want for the key. If I split `serializeMember` into two pieces: one dealing only in values, and one dealing with JsonValueWriter, then this could work. Not sure if it's worth the effort though.